hq/app/logic/SnykDisplay.scala

```diff
@@ -2,63 +2,60 @@ package logic
 
 import com.fasterxml.jackson.core.JsonParseException
 import play.api.libs.json._
-import utils.attempt.{Attempt, FailedAttempt, Failure}
+import utils.attempt.{Attempt, Failure}
 
 import scala.concurrent.ExecutionContext
 import model._
 import model.Serializers._
 
 object SnykDisplay {
 
-  def getOrganisation(s: String, requiredOrganisation: SnykOrganisationName): Attempt[SnykOrganisation] = {
-    val organisation = for {
-      orglist <- parseJsonToOrganisationList(s)
-      org <- orglist.find(_.name == requiredOrganisation.value)
-    } yield org
-    val error = parseJsonToError(s)
-    val f = Failure(s"Unable to find organisation from $s", s"Could not read Snyk response (${error.error})", 502, None, None)
-    Attempt.fromOption(organisation, FailedAttempt(f))
+  def parseOrganisations(s: String, snykGroupId: SnykGroupId)(implicit ec: ExecutionContext): Attempt[List[SnykOrganisation]] = {
+    for {
+      organisationList <- parseJsonToOrganisationList(s)
+      guardianOrganisationList = organisationList filter {
+        case SnykOrganisation(_, id, Some(group)) => group.id==snykGroupId.value
+        case _ => false
+      }
+    } yield guardianOrganisationList
+
   }
 
   private def parseJsonToError(s: String): SnykError = try {
     Json.parse(s).asOpt[SnykError].getOrElse(SnykError(s))
   } catch {
-    case _: JsonParseException => SnykError(s)
+    case e: JsonParseException => SnykError(e.getMessage)
   }
 
-  private def parseJsonToOrganisationList(s: String): Option[List[SnykOrganisation]] = try {
-    (Json.parse(s) \ "orgs").asOpt[List[SnykOrganisation]]
+  def parseJsonToObject[A](label: String, s: String, f: (String => A)): Attempt[A] = try {
+    Attempt.Right(f(s))
   } catch {
-    case _: JsonParseException => None
+    case e: JsonParseException =>
+      val f = Failure(s"Unable to find $label from $s", s"Could not read Snyk response ($s)", 502, None, Some(e))
+      Attempt.Left(f)
+    case e: Exception =>
```

Comment:
`NonFatal` is better than `Exception`, but if you're checking for expected exceptions those are known ahead of time.

The Play JSON library has a `validate` method that returns an either. You'll probably find that easier than using exception handling for the conversion from `JsValue -> A`.
https://www.playframework.com/documentation/2.6.x/ScalaJson#Using-validation

Reply:
Something like this jumps to mind if you want to do it generally for re-use (not compiled)

```scala
def extractJson[A](json: Json)(implicit decoder: Reads[A]): Attempt[A] = {
  Attempt.fromEither(json.validate[A].left.map { jsError =>
    FailedAttempt {
      jsError.errors.map { case (path, errs) =>
        Failure(  ..  do something helpful with the path and errors (use context for path)  ..  )
      }
    }
  })
}
```

There is some relevant stuff in Amiable:
https://github.com/guardian/amiable/blob/master/app/prism/JsonUtils.scala